plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -1410,6 +1413,22 @@ public Optional<ConnectorOutputMetadata> finishInsert(
         return Optional.empty();
     }
 
+    private static List<String> getWritePartitionColumnNames(List<String> originalPartitionColumns, List<DeltaLakeColumnHandle> dataColumns)
+    {
+        return originalPartitionColumns.stream()
+                .map(columnName -> {
+                    DeltaLakeColumnHandle dataColumn = dataColumns.stream()
+                            .filter(column -> columnName.equalsIgnoreCase(column.getName()))
+                            .collect(onlyElement());
+                    // Note: during writes we want to preserve original case of partition columns, if the column's name is not differ of column's physical name
```

Comment:
nit: there is no need for the "Note:" prefix in the statement. The comment implies that this is a developer note.

If we add the column mapping as a parameter for the method, we may not need this comment anymore.
If the column mapping is `NONE` perform the mapping of the column names as in the original code, otherwise for `NAME` use physical column name and for `ID` throw illegal argument exception.